apis/helpers.go

```diff
@@ -92,3 +97,21 @@ func SetDefaultResourceLimits(req *core.ResourceRequirements, defaultResources c
 		req.Requests[r] = calRequest(r, defaultResources.Requests[r])
 	}
 }
+
+func ImageWithDigest(image string) (string, error) {
+	// Drop the "@sha256:hash_string" part, if any
+	if strings.Contains(image, "@") {
+		im := strings.Split(image, "@")
+		if len(im) < 2 || len(im[0]) == 0 {
+			return "", fmt.Errorf("invalid image: %s", image)
+		}
+		image = im[0]
+	}
+
+	digest, err := crane.Digest(image, crane.WithAuthFromKeychain(authn.DefaultKeychain))
```

Comment:
As this is a general library function, so, I think we can take crane digest option as function parameter like `ImageWithDigest(image string, opt crane.Option)`. Then we can pass that opt in crane.Digest func. I believe it will also be helpful when writing unit test against this func.